src/main/js/pullRequest.ts

```diff
@@ -39,27 +143,277 @@ export function createPullRequest(url: string, pullRequest: BasicPullRequest) {
       return apiClient.get(location + "?fields=id");
     })
     .then(response => response.json())
-    .then(pr => pr.id);
+    .then(pr => {
+      if (callback) {
+        callback(pr.id);
+      }
+      return pr;
+    });
 }
 
-export function updatePullRequest(url: string, pullRequest: PullRequest) {
-  return apiClient.put(url, pullRequest, CONTENT_TYPE_PULLREQUEST);
+export const useRejectPullRequest = (repository: Repository, pullRequest: PullRequest) => {
+  const queryClient = useQueryClient();
+  const { mutate, isLoading, error } = useMutation<{}, Error, PullRequest>(pr => rejectPullRequest(pr), {
+    onSuccess: () => {
+      return queryClient.invalidateQueries(prQueryKey(repository, pullRequest.id!));
+    }
+  });
+  return {
+    reject: (pr: PullRequest) => mutate(pr),
+    isLoading,
+    error
+  };
+};
+
+function rejectPullRequest(pullRequest: PullRequest) {
+  return apiClient.post((pullRequest._links.reject as Link).href, {});
 }
 
-export function createPullRequestComment(url: string, comment: BasicComment) {
-  return apiClient.post(url, comment).then(response => {
-    return response;
+type MergeRequest = {
+  url: string;
+  mergeCommit: MergeCommit;
+};
+
+export const useMergePullRequest = (repository: Repository, pullRequest: PullRequest) => {
+  const queryClient = useQueryClient();
+  const { mutate, isLoading, error } = useMutation<{}, Error, MergeRequest>(
+    request => apiClient.post(request.url, request.mergeCommit, "application/vnd.scmm-mergeCommand+json"),
+    {
+      onSuccess: () => {
+        return queryClient.invalidateQueries(prQueryKey(repository, pullRequest.id!));
+      }
+    }
+  );
+  return {
+    merge: (request: MergeRequest) => mutate(request),
+    isLoading,
+    error
+  };
+};
+
+export const usePullRequests = (repository: Repository, status: string) => {
+  const { error, isLoading, data } = useQuery<PullRequestCollection, Error>(
+    ["repository", repository.namespace, repository.name, "pull-requests", status],
+    () =>
+      apiClient
+        .get((repository._links.pullRequest as Link).href + "?status=" + status)
+        .then(response => response.json())
+  );
+
+  return {
+    error,
+    isLoading,
+    data
+  };
+};
+
+export const useDeletePullRequestComment = (repository: Repository, pullRequest: PullRequest) => {
```

Comment:
The review plugin is all about pull requests ...

```suggestion
export const useDeleteComment = (repository: Repository, pullRequest: PullRequest) => {
```